xdsl/parser.py

```diff
@@ -838,43 +815,38 @@ def try_parse_numerical_dims(self,
 
     def parse_vector_attrs(self) -> AnyVectorType:
         # Also break on 'x' characters as they are separators in dimension parameters
-        with self.tokenizer.configured(break_on=self.tokenizer.break_on +
-                                       ("x", )):
-            shape = list[int](self.try_parse_numerical_dims())
-            scaling_shape: list[int] | None = None
-
-            if self.tokenizer.next_token_of_pattern("[") is not None:
-                # We now need to parse the scalable dimensions
-                scaling_shape = list(self.try_parse_numerical_dims())
-                self.parse_characters(
-                    "]", "Expected end of scalable vector dimensions here!")
-                self.parse_characters(
-                    "x", "Expected end of scalable vector dimensions here!")
+        shape = list[int](self.try_parse_numerical_dims())
+        scaling_shape: list[int] | None = None
 
-            if scaling_shape is not None:
-                # TODO: handle scaling vectors!
-                self.raise_error("Warning: scaling vectors not supported!")
-                pass
+        if self.tokenizer.next_token_of_pattern("[") is not None:
+            # We now need to parse the scalable dimensions
+            scaling_shape = list(self.try_parse_numerical_dims())
+            self.parse_characters(
+                "]", "Expected end of scalable vector dimensions here!")
+            self.parse_characters(
+                "x", "Expected end of scalable vector dimensions here!")
 
-            type = self.try_parse_type()
-            if type is None:
-                self.raise_error(
-                    "Expected a type at the end of the vector parameters!")
+        if scaling_shape is not None:
+            # TODO: handle scaling vectors!
+            self.raise_error("Warning: scaling vectors not supported!")
+            pass
+
+        type = self.try_parse_type()
+        if type is None:
+            self.raise_error(
+                "Expected a type at the end of the vector parameters!")
 
-            return VectorType.from_element_type_and_shape(type, shape)
+        return VectorType.from_element_type_and_shape(type, shape)
 
     def _parse_tensor_or_memref_dims(self) -> list[int] | None:
-        with self.tokenizer.configured(break_on=self.tokenizer.break_on +
```

Comment:
Interesting, I though for sure that we'd need this extra break on character here. I guess not!

That would allow us to do some major cleanup! Lovely!

Reply:
My understanding is that the reason why this was working in the first place, was that you are using `next_token_of_pattern`, which was bypassing the `break_on` in the first place!